packages/mgt-components/src/components/mgt-file-list/mgt-file-list.ts

```diff
@@ -0,0 +1,606 @@
+  arraysAreEqual,
+  GraphPageIterator,
+  MgtTemplatedComponent,
+  Providers,
+  ProviderState
+} from '@microsoft/mgt-element';
+import * as MicrosoftGraph from '@microsoft/microsoft-graph-types';
+import { customElement, html, internalProperty, property, TemplateResult } from 'lit-element';
+import { repeat } from 'lit-html/directives/repeat';
+import {
+  getDriveFilesByIdIterator,
+  getDriveFilesByPathIterator,
+  getFilesByIdIterator,
+  getFilesByListQueryIterator,
+  getFilesByPathIterator,
+  getFilesByQueries,
+  getFilesIterator,
+  getGroupFilesByIdIterator,
+  getGroupFilesByPathIterator,
+  getMyInsightsFiles,
+  getSiteFilesByIdIterator,
+  getSiteFilesByPathIterator,
+  getUserFilesByIdIterator,
+  getUserFilesByPathIterator,
+  getUserInsightsFiles
+} from '../../graph/graph.files';
+import '../sub-components/mgt-spinner/mgt-spinner';
+import { OfficeGraphInsightString, ViewType } from '../../graph/types';
+import { styles } from './mgt-file-list-css';
+
+/**
+ * The File List component displays a list of multiple folders and files by
+ * using the file/folder name, an icon, and other properties specicified by the developer.
+ * This component uses the mgt-file component.
+ *
+ * @export
+ * @class MgtFileList
+ * @extends {MgtTemplatedComponent}
+ *
+ * @cssprop
+ *
+ */
+
+@customElement('mgt-file-list')
+export class MgtFileList extends MgtTemplatedComponent {
+  /**
+   * Array of styles to apply to the element. The styles should be defined
+   * using the `css` tag function.
+   */
+  static get styles() {
+    return styles;
+  }
+
+  /**
+   * allows developer to provide query for a file list
+   *
+   * @type {string}
+   * @memberof MgtFileList
+   */
+  @property({
+    attribute: 'file-list-query'
+  })
+  public get fileListQuery(): string {
+    return this._fileListQuery;
+  }
+  public set fileListQuery(value: string) {
+    if (value === this._fileListQuery) {
+      return;
+    }
+
+    this._fileListQuery = value;
+    this.requestStateUpdate();
+  }
+
+  /**
+   * allows developer to provide an array of file queries
+   *
+   * @type {string[]}
+   * @memberof MgtFileList
+   */
+  @property({
+    attribute: 'file-queries',
+    converter: (value, type) => {
+      return value.split(',').map(v => v.trim());
+    }
+  })
+  public get fileQueries(): string[] {
+    return this._fileQueries;
+  }
+  public set fileQueries(value: string[]) {
+    if (arraysAreEqual(this._fileQueries, value)) {
+      return;
+    }
+
+    this._fileQueries = value;
+    this.requestStateUpdate();
+  }
+
+  /**
+   * allows developer to provide an array of files
+   *
+   * @type {MicrosoftGraph.DriveItem[]}
+   * @memberof MgtFileList
+   */
+  @property({ type: Object })
+  public get files(): MicrosoftGraph.DriveItem[] {
+    return this._files;
+  }
+  public set files(value: MicrosoftGraph.DriveItem[]) {
+    if (value === this._files) {
+      return;
+    }
+
+    this._files = value;
+    this.requestStateUpdate();
```

Comment:
You don't need to call `requestStateUpdate` here (in fact, you should get rid of the getter/setter for `files`) since changing this prop doesn't require state update (calling the graph). This will cause the the `loadState` method to be called each time the files are set.

Reply:
What if a developer wants to set files from js? Does it update correctly without a getter/setter?

Reply:
It does, the `@property` decorator autogenerates the setters and getters if you don't provide them